### PR TITLE
300 bitfields wrapping

### PR DESCRIPTION
**Pull request by uhetmaniuk**
300 bitfields wrapping

### CODE REVIEW

File: src/vt/group/group_manager.cc [hunk 1]
```diff
@@ -83,7 +83,11 @@ GroupType GroupManager::newGroupCollective(
 
 GroupType GroupManager::newGroupCollectiveLabel(GroupCollectiveLabelTagType) {
   auto const& this_node = theContext()->getNode();
-  auto next_id = next_collective_group_id_++;
+  auto const next_id = next_collective_group_id_;
+  vt::utils::FieldWrapper<
```
Comment (pnstickne) [inline src/vt/group/group_manager.cc]: I like how this block is formatted (and without the //) - make the others to match?

File: src/vt/handler/handler.h [hunk 2]
```diff
@@ -62,7 +62,8 @@ using HandlerIdentifierType = uint64_t;
 using HandlerControlType = uint32_t;
 
 static constexpr HandlerIdentifierType const first_handle_identifier = 1;
-static constexpr HandlerIdentifierType const uninitialized_handle_identifier = -1;
+static constexpr HandlerIdentifierType const uninitialized_handle_identifier
```
Comment (pnstickne) [inline src/vt/handler/handler.h]: Whoa lines!

File: src/vt/registry/auto/auto_registry_common.h [hunk 3]
```diff
@@ -78,6 +78,8 @@ using HandlerManagerType = vt::HandlerManager;
 using AutoHandlerType = int32_t;
 using NumArgsType = int16_t;
 
+static constexpr AutoHandlerType const not_set_auto_handler = static_cast<AutoHandlerType>(-1);
```
Comment (pnstickne) [inline src/vt/registry/auto/auto_registry_common.h]: Why is this one special compared to the others?

File: src/vt/utils/wrapper/field_wrapper.h [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+
+template< typename T, typename BitCountType >
+constexpr T maxWrap(BitCountType len) {
+  return (len == 1) ? 0 : (maxWrap<T,BitCountType>(len-1) << 1) + 1;
+};
+
+
+template<fieldName wrapField, typename FieldType, BitCountType len>
+struct FieldWrapper
```
Comment (pnstickne) [inline src/vt/utils/wrapper/field_wrapper.h]: What is overall advantage/goal of wrapped fields?

File: src/vt/utils/wrapper/field_wrapper.h [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+#include <unordered_set>
+
+
+#include "vt/configs/types/types_type.h"
+
+
+namespace vt { namespace utils {
+
+
+enum struct fieldName {
```
Comment (pnstickne) [inline src/vt/utils/wrapper/field_wrapper.h]: Meaning for this being `fieldName` instead of `FieldName`?

**Comment by cz4rs** [comment]
Closing as there is no urgence to get this updated and merged.
Note: this might contain valid ideas for fixing the issue.

**Pull request closed without merging**